python/pyspark/resource/executorresourcerequest.py

```diff
@@ -0,0 +1,73 @@
+#
```

Comment:
@tgravescs, can we just merge all these file into one `resource` file? It's not very common for one file to have one class in Python side up to my knowledge.

Reply:
I was basing this off what i saw in other files for external api.  TaskContext, Broadcast, RDD, accumulator, sql/Catalog, sql/Dataframe, etc.  Can you tell me what this fits with if not those?

Reply:
Yes, I noticed some files are like that but the classes here look pretty small. There are opposite examples such as `pyspark/sql/readwriter.py`, `pyspark/sql/streaming.py`, `pyspark/status.py`, `pyspark/sql/window.py`, `pyspark/sql/utils.py`, `pyspark/sql/udf.py` and `pyspark/sql/types.py`. I admit there's no strict rule on this and I have no point of the official documentation to follow.

Just in practice, it's usual to group several relevant classes, e.g.) https://stackoverflow.com/questions/106896/how-many-classes-should-i-put-in-one-file. Python module is more like Java's package instead of class file.

Reply:
I don't think utils is a good example, utils files are generally a bunch of things. Most of the files you mention have 2 classes in them. It seems to be personal preference.  So if I make a change, the question is do I put all 6 classes into 1 file, or do I split up by executor, task, and resourceprofile/builder in 3 different files.  Again if I'm going to make a change I would like to be clear to not drag this out and have to rework things multiple times.

@holdenk any input on this?

Reply:
I would have not commented if this was one class per one file. My point is to group the classes into one or several files. You can group it as you want but one class per on file seems not so "Pythonic" as I pointed out with the links and examples.